### PR TITLE
NAS-110653 / 21.06 / NAS-110653: Improving type safety of Sharing and ACL

### PR DESCRIPTION
**Pull request by undsoft**
Mostly type changes and minor refactoring. Don't think it's worth testing.

### CODE REVIEW

File: src/app/pages/account/users/user-form/user-form.component.ts [hunk 1]
```diff
@@ -356,16 +356,18 @@ export class UserFormComponent implements FormConfiguration {
...
-      // On a new form, if there is a home SMB share, populate the 'home' form explorer with it...
-        .subscribe((res) => {
-          if (res.length > 0) {
-            this.homeSharePath = res[0].path;
-            this.entityForm.formGroup.controls['home'].setValue(this.homeSharePath);
-            // ...then add on /<username>
-            this.entityForm.formGroup.controls['username'].valueChanges.subscribe((value: string) => {
-              this.entityForm.formGroup.controls['home'].setValue(`${this.homeSharePath}/${value}`);
-            });
+        .subscribe((shares) => {
```
Comment (RehanY147) [inline src/app/pages/account/users/user-form/user-form.component.ts:359]: Should we have a type for `shares` here?

File: src/app/pages/services/components/service-smb/service-smb.component.ts [hunk 2]
```diff
@@ -299,20 +300,20 @@ export class ServiceSMBComponent implements FormConfiguration {
       }
     });
 
-    this.ws.call('user.query').subscribe((res) => {
+    this.ws.call('user.query').subscribe((users) => {
```
Comment (RehanY147) [inline src/app/pages/services/components/service-smb/service-smb.component.ts:303]: Not getting type intellisense for `users` here

File: src/app/pages/storage/volumes/datasets/dataset-posix-acl/dataset-posix-acl.component.ts [hunk 3]
```diff
@@ -275,22 +276,22 @@ export class DatasetPosixAclComponent implements FormConfiguration, OnDestroy {
       });
     });
 
-    this.userService.userQueryDSCache().subscribe((items) => {
-      const users = [];
-      for (let i = 0; i < items.length; i++) {
-        users.push({ label: items[i].username, value: items[i].username });
+    this.userService.userQueryDSCache().subscribe((users) => {
```
Comment (RehanY147) [inline src/app/pages/storage/volumes/datasets/dataset-posix-acl/dataset-posix-acl.component.ts:279]: No intellisense for type of `Users[]`

**Review by RehanY147 — APPROVED** [review]

File: src/app/pages/storage/volumes/datasets/dataset-posix-acl/dataset-posix-acl.component.ts (hunk 3 above)
Comment (undsoft) [reply src/app/pages/storage/volumes/datasets/dataset-posix-acl/dataset-posix-acl.component.ts:279]: Seems to be a local IDE issue.

**Pull request merged**